tasks/check.py

```diff
@@ -0,0 +1,88 @@
+import compileall
+import glob
+import os
+import re
+
+from invoke import run, task
+
+from . import copy
+
+
+@task
+def compile_(ctx):
+    print("======================= compile ========================")
+    print("------- Compile all .py files (syntax check test) ------")
+    compileall.compile_dir(".", rx=re.compile(r"/virtualenv|virtualenv-osx|virtualenv-py3|.tox|.git|.venv-st2devbox"), quiet=True)
```

Comment:
Will this correctly throw and exit with non zero if compile_dir fails?

Reply:
EDIT:

> Recursively descend the directory tree named by dir, compiling all .py files along the way. Return a true value if all the files compiled successfully, and a false value otherwise.

https://docs.python.org/3/library/compileall.html#compileall.compile_dir

So it looks like we may want to check the return value and throw / exit with non zero if this function returns False?